source/tutorials/nix-language.md

```diff
@@ -0,0 +1,325 @@
+# Reading the Nix language
+
+The Nix language is used to declare packages and configurations for the Nix package manager.
+
+You will quickly encounter Nix language expressions that may look very complicated.
+Yet, the language has only few basic concepts that can be combined arbitrarily.
+Use this guide to learn reading code in Nix language and understand its structure.
+
+Imagine the Nix language as JSON with functions.
+Its purpose is to define structured data, and functions help with conveniently producing more complex data.
+All of the language constructs can be composed arbitrarily.
+
+### Attribute sets
+
+The primary data structure is the attribute set, a collection of key-value pairs.
```

Comment:
> Calling it the primary data structure feels weird

I wanted to say it's the most important one, because everything revolves around attrsets. See [JSON+lambdas](https://github.com/NixOS/nix.dev/pull/267/files#r932169165). 

> Also data structure is a bit off

Could you clarify why, just so I understand?

(In general I agree that using the most precise terms is more correct. Yet, I don't think its also more better, as the goal of these tutorials is to pick up developers, who are not necessarily all too familiar with PLT, from where they are). 

> better distinction between values and expressions. This document here mixes these two concepts fairly arbitrarily.

Could you please provide concrete suggestions?